qiskit/ignis/verification/randomized_benchmarking/dihedral.py

```diff
@@ -104,18 +100,15 @@ def __mul__(self, other):
         result = SpecialPolynomial(self.n_vars)
         if isinstance(other, int):
             result.weight_0 = (self.weight_0 * other) % 8
-            result.weight_1 = [(x * other) % 8 for x in self.weight_1]
-            result.weight_2 = [(x * other) % 8 for x in self.weight_2]
-            result.weight_3 = [(x * other) % 8 for x in self.weight_3]
+            result.weight_1 = (((np.array(self.weight_1) * other) % 8).astype(int)).tolist()
```

Comment:
Do we really need to convert back to list? Most of the type the Python code works just as well on `np.array`